internal/domain/application_oidc.go

```diff
@@ -386,24 +380,23 @@ func containsCustom(uris []string) bool {
 
 func onlyLocalhostIsHttp(uris []string) bool {
 	for _, uri := range uris {
-		if strings.HasPrefix(uri, http) && !isHTTPLoopbackLocalhost(uri) {
+		if url, err := url.ParseRequestURI(uri); err == nil && url.Scheme == "http" {
+			hostname := url.Hostname()
+
+			if hostname == "localhost" {
+				continue
+			}
+
+			if address, err := netip.ParseAddr(hostname); err == nil && address.IsLoopback() {
```

Comment:
Same here, if `err != nil` this will return `false`.

I understand that at this line of code, we're in the scope of _url starts with http_ and we're trying to assess whether it's a localhost IP or not.

but the fact that hostname cannot be parsed, does it qualify `address` as a valid one in the first place?

I'd speculate that if `ParseRequestURI()` succeeds, it's also because `hostname` is valid, so the error here could be ignored.